vault/plugin_catalog.go

```diff
@@ -237,8 +243,6 @@ func (c *PluginCatalog) getPluginClient(ctx context.Context, sys pluginutil.Runn
 
```

Comment:
This may be an assumption about how the go-plugin code behaves today but may not be an explicit guarantee of the API? It may be safer to instead cache a single instance of this on the mpc object?